pandas/indexes/category.py

```diff
@@ -136,6 +136,19 @@ def _simple_new(cls, values, name=None, categories=None, ordered=None,
         result._reset_identity()
         return result
 
+    @Appender(_index_shared_docs['_shallow_copy'])
+    def _shallow_copy(self, values=None, categories=None, ordered=None,
+                      **kwargs):
+        # categories and ordered can't be part of attributes,
+        # as these are properties
+        if categories is None:
```

Comment:
Based on the current rules, shouldn't ``_shallow_copy`` be used if metadata of caller is needed?

- https://github.com/pydata/pandas/blob/master/pandas/indexes/base.py#L287

> - _simple_new: It returns new Index with the same type as the caller. All metadata (such as name) must be provided by caller's responsibility. Using _shallow_copy is recommended because it fills these metadata otherwise specified.
> - _shallow_copy: It returns new Index with the same type (using _simple_new), but fills caller's metadata otherwise specified. Passed kwargs will overwrite corresponding metadata.